core/src/win32/plugins/filed/mssqlvdi-fd.cc

```diff
@@ -1373,18 +1406,54 @@ static inline bool SetupVdiDevice(bpContext* ctx, struct io_pkt* io)
    */
   p_ctx->AdoThreadStarted = true;
 
-  /*
-   * Wait for the database server to connect to the VDI deviceset.
-   */
-  hr = p_ctx->VDIDeviceSet->GetConfiguration(VDI_DEFAULT_WAIT,
-                                             &p_ctx->VDIConfig);
-  if (!SUCCEEDED(hr)) {
-    Jmsg(ctx, M_FATAL,
-         "mssqlvdi-fd: IClientVirtualDeviceSet2::GetConfiguration failed\n");
-    Dmsg(ctx, debuglevel,
-         "mssqlvdi-fd: IClientVirtualDeviceSet2::GetConfiguration failed\n");
-    goto bail_out;
-  }
+  // GetConfiguration
+  {
+    uint32_t timeout = p_ctx->get_configuration_timeout == 0
+                           ? VDI_DEFAULT_WAIT
+                           : p_ctx->get_configuration_timeout * 1000;  // ms
+
+    Jmsg(ctx, M_INFO, "Calling GetConfiguration with a timeout of %d sec\n.",
+         timeout / 1000);
+
+    hr = p_ctx->VDIDeviceSet->GetConfiguration(timeout, &p_ctx->VDIConfig);
+
+    const char* err = "";
+    bool success = false;
+
+    switch (hr) {
+      case NOERROR:
+        err = "NOERROR";
+        success = true;
+        break;
+      case VD_E_ABORT:
+        err = "VD_E_ABORT";
+        break;
+      case VD_E_TIMEOUT:
+        err = "VD_E_TIMEOUT, you may want to increase getconfigurationtimeout";
+        break;
+      default:
+        err = "unspecific error";
+        break;
+    }
+
+    std::array<char, 300> error_msg;
+    const char* fmt =
+        "mssqlvdi-fd: IClientVirtualDeviceSet2::GetConfiguration "
+        "%s: %0#x (%s)\n";
+
+    if (success) {
+      sprintf(error_msg.data(), fmt, "successful",
```

Comment:
would the compiler warn if the destination buffer is too short? The possible values are all known at compile-time, so there should be a warning.
I'd consider using `snprintf()` if the compiler doesn't warn, as it makes the code more robust to changes.